src/order/directed.lean

```diff
@@ -189,6 +214,27 @@ lemma is_bot_iff_is_min [is_directed α (≥)] : is_bot a ↔ is_min a :=
 
 lemma is_top_iff_is_max [is_directed α (≤)] : is_top a ↔ is_max a := ⟨is_top.is_max, is_max.is_top⟩
 
+/-- The property that a function preserves `is_lub` on directed sets. -/
+def preserves_lub_on_directed [preorder β] (f : α → β) : Prop :=
+∀ (d : set α) (a : α), d.nonempty → directed_on (≤) d → is_lub d a → is_lub (f '' d) (f a)
```

Comment:
Is there not a better name coming from the literature?

Reply:
The term [Scott Continuous](https://en.wikipedia.org/wiki/Scott_continuity) is used - I show [here](https://github.com/leanprover-community/mathlib4/pull/2508/files#diff-983e16dd03108ad09234625f776ad94cfb58b1f6ee4762bf7519306fd90a2b46R294) that the functions satisfying this property are exactly those which are continuous wrt the Scott topology.

Reply:
This is the approach taken [here](https://github.com/leanprover-community/mathlib/blob/master/src/order/omega_complete_partial_order.lean#L233). Personally I like to establish the topology before I talk about functions being continuous, but I appreciate it is possible to do it the other way around.

Reply:
I've renamed it to `scott_continuous`.